routing/datasource.go

```diff
@@ -353,48 +353,73 @@ func mapPredicates(cps []PredicateSpec) map[string]PredicateSpec {
 }
 
 // processes a set of route definitions for the routing table
-func processRouteDefs(o Options, fr filters.Registry, defs []*eskip.Route) []*Route {
+func processRouteDefs(o Options, fr filters.Registry, defs []*eskip.Route) (routes []*Route, invalidDefs []*eskip.Route) {
 	cpm := mapPredicates(o.Predicates)
-
-	var routes []*Route
 	for _, def := range defs {
 		route, err := processRouteDef(cpm, fr, def)
 		if err == nil {
 			routes = append(routes, route)
 		} else {
+			invalidDefs = append(invalidDefs, def)
 			o.Log.Error(err)
 		}
 	}
+	return
+}
 
-	return routes
+type routeTable struct {
+	m             *matcher
+	validRoutes   []*Route
+	invalidRoutes []*Route
```

Comment:
i think both the validRoutes and the invalidRoutes should be only the slice of the eskip.Route type. Invalid routes and invalidDefs could be a single slice